src/shell/execute.rs

```diff
@@ -904,6 +904,8 @@ fn evaluate_word_parts(
             None
           }
           WordPart::Variable(name) => state.get_var(OsStr::new(&name)).cloned(),
+          WordPart::Tilde => sys_traits::impls::real_home_dir_with_env(state)
+            .map(|s| s.into_os_string()),
```

Comment:
I was just thinking... maybe we should error when this is `None`? I feel like it could be very surprising or cause destruction/bad behaviour if not.

Reply:
Yeah, made this change.